src/core/iterator.ml

```diff
@@ -0,0 +1,41 @@
+
+type +'a node =
```

Comment:
Can we rename this module Sequence, include the Sequence type definition in core, but move your implementation of the combinators into containers.